migration-engine/core/src/migration/datamodel_migration_steps_inferrer.rs

```diff
@@ -43,6 +45,8 @@ impl<'a> DataModelMigrationStepsInferrerImpl<'a> {
         result.append(&mut Self::wrap_as_step(enums_to_create, MigrationStep::CreateEnum));
         result.append(&mut Self::wrap_as_step(enums_to_delete, MigrationStep::DeleteEnum));
         result.append(&mut Self::wrap_as_step(enums_to_update, MigrationStep::UpdateEnum));
+        result.append(&mut Self::wrap_as_step(indexes_to_create, MigrationStep::CreateIndex));
+        result.append(&mut Self::wrap_as_step(indexes_to_delete, MigrationStep::DeleteIndex));
```

Comment:
I have the feeling that it would be safer to delete indexes before creating the new ones. There might be obscure edge cases where the creation might fail because other indexes have not been deleted yet.

Reply:
Fixed in 39e77849da95ef4ec51d8809cdfe681c41e90870